protocol-models/src/main/resources/airbyte_protocol/v0/airbyte_protocol.yaml

```diff
@@ -485,6 +487,15 @@ definitions:
           type: array
           items:
             type: string
+      generation_id:
+        description: Monotically increasing numeric id representing the current generation of a stream. This id can be shared across syncs
+        type: integer
+      minimum_generation_id:
+        description: The minimum generation id which is needed in a stream. If it is present, the destination will try to delete the data that are part of a generation lower than this property. If it is absent, the destination won't try to delete data
+        type: integer
```

Comment:
Why would this be null?  I think on the first sync both generation_id and minimum_generation_id both be "0", and non-null

Reply:
This is set to null for regular sync where clearing the data is not needed. Having the minimum generation id won't be useful for the destination in this case.

Reply:
Is it a problem if we always include it? Allowing null implies a 3rd state that is a little confusing. Unless there is a reason we must allow it to be nullable, I think it makes most sense to always set it.

Reply:
No, the behavior if we set it to 0 will be the same than setting it at null. It will mean that we will not perform a cleaning of the data. FYI @edgao

Reply:
kk. if that's the case. i agree with evan. it shouldn't be nullable.